tests/unit/Entity/EntityIdValueTest.php

```diff
@@ -75,41 +78,95 @@ public function testGetSortKey( EntityIdValue $id ) {
 		$this->assertInternalType( 'string', $id->getSortKey() );
 	}
 
+	public function provideGetArrayValue() {
+		return [
+			'Q2147483647' => [
+				new ItemId( 'Q2147483647' ),
+				[
+					'entity-type' => 'item',
+					'numeric-id' => 2147483647,
+					'id' => 'Q2147483647'
+				],
+			],
+			'P31337' => [
+				new PropertyId( 'P31337' ),
+				[
+					'entity-type' => 'property',
+					'numeric-id' => 31337,
+					'id' => 'P31337',
+				],
+			],
+			'X567' => [
+				$this->newCustomId( 'X567' ),
+				[
+					'entity-type' => 'custom',
+					'id' => 'X567',
+				],
+			],
+			'foo:P678' => [
+				new PropertyId( 'foo:P678' ),
+				[
+					'entity-type' => 'property',
+					'numeric-id' => 678,
+					'id' => 'foo:P678',
+				],
+			],
+		];
+	}
+
 	/**
-	 * @dataProvider instanceProvider
+	 * @dataProvider provideGetArrayValue
 	 */
-	public function testGetArrayValueRoundtrip( EntityIdValue $id ) {
-		$newId = EntityIdValue::newFromArray( $id->getArrayValue() );
+	public function testGetArrayValue( EntityId $id, array $expected ) {
+		$value = new EntityIdValue( $id );
+		$array = $value->getArrayValue();
 
-		$this->assertEquals( $id, $newId );
+		ksort( $expected );
+		ksort( $array );
+		$this->assertEquals( $expected, $array );
```

Comment:
> […] makes calling code more brittle, and the code itself harder to maintain.

I hope you realize that I just said the exact opposite?

> […] not stated anywhere

The guarantee is "stated" as a fact in every single serialization we export.

> […] which no production code relies on (I hope)?

I also hope no production and no third party code relies on this. This does not mean it's "pointless" to give guarantees. It's simply convenient for developers to be able to look at an export and always find the keys in the same order.

Fixing this is super trivial:
```php
$array = [ 'first key' => … ];
if ( … ) {
    $array['second key'] = …;
}
$array['third key'] = …;
```